plugin/src/main/java/io/fabric8/maven/plugin/mojo/AbstractFabric8Mojo.java

```diff
@@ -49,6 +50,97 @@
     @Parameter(defaultValue = "${settings}", readonly = true)
     protected Settings settings;
 
+
+    /**
+     * Namespace on which to operate
+     */
+    @Parameter(property = "fabric8.namespace")
+    private String namespace;
+
+    /**
+     * Master URL on which to operate
+     */
+    @Parameter(property = "fabric8.masterUrl")
+    private String masterUrl;
+
+    /**
+     * Api version on which to operate
+     */
+    @Parameter(property = "fabric8.apiVersion")
```

Comment:
Couldn't we use `ClusterConfiguration` directly as property so that we would have nested configs like

```xml
<configuration>
  <access>
    <apiVersion>...</apiVersion>
    <caCertData>....</caCertData>
    ....
  </access>
</configuration>
```

This would reduce the noise in the `AbstractDockerMojo`. The drawback would be that we would have to evaluate the properties on our own. However in this case we would be even able to use both Maven properties *and* system properties (which is not the case when using @Parameter)

Reply:
I thought about it but notice that this will break back compatibility (`namespace` field) was already there before this change.

Reply:
We can keep the namespace field, mark it is deprecated and if set, used it to fill in the ClusterConfiguration's namespace field if this is set (but not the one in ClusterConfiguration). That we, still can support the old namespace part for some time and remove it eventually. 

Not sure so, as we are moving to 4.0 anyway, whether we just should do the switch and note this as a migration step. 

wdyt ?

Reply:
ok, then let's remove the field, I prefer to remove it right now instead of maintaining because if we maintain, we will never find the correct time to remove it.